libvast/test/detail/type_traits.cpp

```diff
@@ -8,13 +8,32 @@
 
 #define SUITE type_traits
 
+#include "vast/detail/type_traits.hpp"
+
 #include "vast/test/test.hpp"
 
-#include "vast/detail/type_traits.hpp"
+#include <caf/variant.hpp>
+
+#include <variant>
 
 TEST(sum) {
   static_assert(vast::detail::sum<> == 0);
   static_assert(vast::detail::sum<1, 2, 3> == 6);
   static_assert(vast::detail::sum<42, 58> == 100);
 }
 
+template <class... Ts>
+struct fake_list {};
+
+template <template <class...> class TList>
+constexpr auto check() {
+  static_assert(vast::detail::contains_type_v<TList<int, double>, double>);
+  static_assert(not vast::detail::contains_type_v<TList<int, double>, char>);
+}
```

Comment:
These need to be in anonymous namespaces, otherwise we risk running into (possibly silent) linker errors.